test/unistd/test_unistd_write_broken_link.c

```diff
@@ -10,21 +11,23 @@ int main() {
   printf("link result: %d\n", res);
   int src_fd = open("link_source", O_CREAT | O_WRONLY, 0777);
   printf("source_fd: %d, errno: %d %s\n", src_fd, errno, strerror(errno));
-  write(src_fd, "abc", 3);
+  write(src_fd, "abc\0", 4);
   close(src_fd);
   {
     int target_fd = open("link_target", O_RDONLY);
     printf("target_fd: %d, errno: %d %s\n", target_fd, errno, strerror(errno));
     char buf[10];
-    read(target_fd, buf, 10);
+    size_t r = read(target_fd, buf, 10);
+    assert(r == 4);
     printf("buf: '%s'\n", buf);
     close(target_fd);
   }
   {
     int target_fd = open("link_source", O_RDONLY);
     printf("target_fd: %d, errno: %d %s\n", target_fd, errno, strerror(errno));
     char buf[10];
```

Comment:
An alternative here I suppose would be intialize buf on the stack with `char buf[10] = { '\0' };`

Reply:
Yeah, I could go either way on that, do you have a preference?